### ISSUE TITLE
Error: chown - operation not permitted

### ISSUE THREAD

**Issue by wtfacoconut** [mikefarah/yq#1521]
**Describe the bug**
Version 4.30.7 on Ubuntu produces the following error:
`Error: chown /tmp/temp4010259877: operation not permitted`

Note that any how to questions should be posted in the discussion board and not raised as an issue.

Version of yq: 4.30.7
Operating system: Linux (Ubuntu 22.04.1 LTS)
Installed via: snap

**Input Yaml**
N/A

**Command**
The command you ran:
```
yq -i '.postgres.postgresUser.enabled = true' sd-local-deps-release-2022.08/sd-local-deps/values.yaml
```

**Additional context**
Using the following command, I downgraded to v4.30.3 where the previously mentioned error does NOT occur:
`sudo snap refresh yq --channel=v4/stable`

**Comment by srheaume** [comment]
This bug should be treated with high priority. It makes the utility completely unusable on Ubuntu

**Comment by mikefarah** [comment]
ah snap - my arch nemesis. Sorry folks, looking into this

**Comment by mikefarah** [comment]
Fixed in 4.30.8
